### ISSUE TITLE
`reportOverlappingOverload` false positive with paramspec and callable `Protocol`

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#1738]
### Description

Code sample in [basedpyright playground](https://basedpyright.com/?pyrightVersion=1.36.0&typeCheckingMode=all&reportUnusedCallResult=false&reportUnusedParameter=false&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoDCAhgDYlEBGJApgDRQAK4MYAxmCfWAG7UglgiAE3ohqvUgH14CagFgAUIsWtyAZzVQAYigDaAKn0N6AFQC6ACiZgW7EgEoAXIqiuoQ6sCiTJrUiR8LNWoSYHpgMDBHTBQYen0iEDQ1aIYAOkTk%2BP0AawB3TJTGNPzC%2BygAWgA%2BKAA5MBRqaLSW5SUFVSINbRQAJitmNg4nFzcPLx8-MkDg0PoKaNQYcuq6hqaoFrS2xQABHj4BYUVxqAiwC2AUaJ1dLfp6xrMVmqXm1va9g-5BIRPPM6RS7XQj%2BSg0O4tB7rZ6VGpqGAgd5pVwAYigfHASKgAHleD9hFBeoCQFAAETnMlQPJIMhQRr4qAUahQACuwSETOofnZLKQME0CESRAg1BgfE033ICCgUt%2BUAAjFALGIcHg8YciAhkOgNQShPYdgpTudgdEwBQAFbc5Zw2VWm3Io2nIiXSLRBEgJybD5tMQSAIyahui5EezldGexRAA)

```python
from typing import Callable, Protocol, overload, reveal_type


class Fn[**P, T](Protocol):
    def __call__(self, foo: int, *args: P.args, **kwargs: P.kwargs) -> None: ...


class Fn2(Protocol):
    def __call__(self, b: int) -> None: ...


@overload
def foo(fn: Fn[..., None]) -> int: ...


@overload
def foo(fn: Callable[..., None]) -> str: ...  # error: Overload 2 for "foo" will never be used because its parameters overlap overload 1 (reportOverlappingOverload)


def foo(fn: object) -> object: ...


def a(foo: str): ...


reveal_type(foo(a)) # str
```

seems very similar to https://github.com/microsoft/pyright/issues/10587
